surfactant/relationships/dotnet_relationship.py

```diff
@@ -65,6 +65,34 @@ def establish_relationships(
                                 dnProbingPaths = []
                             dnProbingPaths.append(pathlib.PureWindowsPath(path).as_posix())
 
+    # https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/loading-unmanaged
+    # 1. Check the active AssemblyLoadContext cache
+    # 2. Calling the import resolver set by the setDllImportResolver function
+    #    - a. Example using SetDllImportResolver: https://learn.microsoft.com/en-us/dotnet/standard/native-interop/native-library-loading
+    #    - b. Checks PInvoke's or Assembly's DefaultDllImportSearchPathsAttribute, then the assembly's directory, then LoadLibraryEx with LOAD_WITH_ALTERED_SEARCH_PATH flag
+    #       - This attribute has no effect on non-Windows platforms / Mono runtime
+    #       - https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.defaultdllimportsearchpathsattribute?view=net-7.0
+    #       - i. This has a "Paths" property which is a bitwise combination of paths specified in ii:
+    #       - ii. https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.dllimportsearchpath?view=net-7.0
+    # 3. The active AssemblyLoadContext calls its LoadUnmanagedDll function (Default behavior is the same as AssemblyRef probing?)
```

Comment:
Based on https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/default-probing#satellite-resource-assembly-probing it seems like it is -- though it's hard to tell if that satellite assembly probing section applies to loading unmanaged libraries as well. It's possible that the default probing logic is just https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/default-probing#unmanaged-native-library-probing and none of the AssemblyRef probing stuff applies.

Reply:
Decided on re-reading the page that satellite-resource-assembly-probing does not apply.